File: stock-serveur-v2/src/main/java/cm/belrose/stockserveur/model/CommandeFournisseur.java
package cm.belrose.stockserveur.model;

import cm.belrose.stockserveur.config.audit.Auditable;
import lombok.*;
import org.hibernate.envers.NotAudited;

import javax.persistence.*;
import java.io.Serializable;
import java.time.Instant;
import java.util.List;

/**
 *
 *Le 09/11/2020
 *
 *@author  Ngnawen Samuel
 *
 */

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "Commandefournisseurs")
public class CommandeFournisseur extends Auditable<String> implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String code;
    @Column(name = "date_commande")
    private Instant dateCommande;
    private EtatCommande etatCommande;
    @NotAudited
    @ManyToOne
    @JoinColumn(name = "fournisseur_id")
    private Fournisseur fournisseur;
    @NotAudited
    @OneToMany(mappedBy = "commandeFournisseur")
    private List<LigneCommandeFournisseur> ligneCommandeFournisseurList;
    //Ce champs est juste mis pour simplifier les choses
    //@Column(name = "entreprise_id")
    private Long entrepriseId;
}

File: stock-client-v2/src/gs-api/src/models/commande-fournisseur.ts
/* tslint:disable */
import { Fournisseur } from './fournisseur';
import { LigneCommandeFournisseur } from './ligne-commande-fournisseur';
export interface CommandeFournisseur {
  createdBy?: string;
  updatedBy?: string;
  createdDate?: string;
  updateDate?: string;
  id?: number;
  code?: string;
  dateCommande?: number;
  etatCommande?: 'LIVREE' | 'VALIDEE' | 'EN_COURS';
  fournisseur?: Fournisseur;
  ligneCommandeFournisseurList?: Array<LigneCommandeFournisseur>;
  entrepriseId?: number;
}
